src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let resultStr = '';
+    let normalizedSeparator;
+
+    if (separator === undefined) {
+      normalizedSeparator = ',';
+    } else {
+      normalizedSeparator = String(separator);
```

Comment:
all concatenations with string will give you `string' result so you can avoid this part as well